broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/LogStreamPartitionTransitionStep.java

```diff
@@ -0,0 +1,108 @@
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import io.atomix.raft.RaftServer.Role;
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionContext;
+import io.camunda.zeebe.broker.system.partitions.PartitionTransitionStep;
+import io.camunda.zeebe.logstreams.log.LogStream;
+import io.camunda.zeebe.logstreams.log.LogStreamBuilder;
+import io.camunda.zeebe.logstreams.storage.atomix.AtomixLogStorage;
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import io.camunda.zeebe.util.sched.future.CompletableActorFuture;
+import java.util.function.Supplier;
+
+public class LogStreamPartitionTransitionStep implements PartitionTransitionStep {
+
+  private final Supplier<LogStreamBuilder> logStreamBuilderSupplier;
+
+  public LogStreamPartitionTransitionStep() {
+    this(LogStream::builder);
+  }
+
+  // Used for testing
+  public LogStreamPartitionTransitionStep(
+      final Supplier<LogStreamBuilder> logStreamBuilderSupplier) {
+    this.logStreamBuilderSupplier = logStreamBuilderSupplier;
+  }
+
+  @Override
+  public ActorFuture<Void> prepareTransition(
+      final PartitionTransitionContext context, final long term, final Role targetRole) {
+    final var logStream = context.getLogStream();
+    if (logStream != null
+        && (shouldInstallOnTransition(targetRole, context.getCurrentRole())
+            || targetRole == Role.INACTIVE)) {
+      context.getComponentHealthMonitor().removeComponent(logStream.getLogName());
+      final ActorFuture<Void> future = logStream.closeAsync();
+      future.onComplete(
+          (ok, error) -> {
+            if (error == null) {
+              context.setLogStream(null);
+            }
+          });
+
+      return future;
+    } else {
+      return CompletableActorFuture.completed(null);
+    }
+  }
+
+  @Override
+  public ActorFuture<Void> transitionTo(
+      final PartitionTransitionContext context, final long term, final Role targetRole) {
+    if ((context.getLogStream() == null && targetRole != Role.INACTIVE)
+        || shouldInstallOnTransition(targetRole, context.getCurrentRole())) {
+      final CompletableActorFuture<Void> openFuture = new CompletableActorFuture<>();
+
+      final var logStorage = context.getLogStorage();
+      buildLogstream(context, logStorage)
+          .onComplete(
+              ((logStream, err) -> {
+                if (err == null) {
+                  context.setLogStream(logStream);
+
+                  context
+                      .getComponentHealthMonitor()
+                      .registerComponent(logStream.getLogName(), logStream);
+                  openFuture.complete(null);
+                } else {
+                  openFuture.completeExceptionally(err);
+                }
+              }));
+
+      return openFuture;
+    } else {
+      return CompletableActorFuture.completed(null);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return "logstream";
```

Comment:
Please have a look across steps? I feel the naming is not consistent. This is lowercase, I think I have also seen camel case before. But I am not sure.

Reply:
I was just reusing code from existing PartitionStep. I can rename everything all at once after all steps are moved.